lib/src/platform/src/platform.dart

```diff
@@ -18,27 +18,29 @@ class Platform {
   static final StreamsChannel streamsChannel =
       StreamsChannel('io.ably.flutter.stream', codec);
 
-  /// Initializing ably on platform side by invoking `register` platform method.
-  /// Register will clear any stale instances on platform.
-  static Future? _initializer;
+  /// This field will reset to true when the state is reset. This allows us to
+  /// if an app has been restarted, or a hot restart (not hot reload) has
+  /// happened.
+  static bool _shouldResetAblyClients = true;
 
-  static Future _initialize() async {
-    if (_initializer == null) {
+  /// Clears instances on the Platform side
+  static Future<void> _resetOldAblyClients() async {
```

Comment:
Sorry to be picky on static methods but are we not performing this methods on Platform object here? Using statics will defeat the purpose of encapsulation and will cause the design to consume other static states which makes application unmaintainable. I think those static methods, fields should be used a bit more cautiously

Reply:
Don't be sorry, its definitely useful to think about this.

Everything in `Platform` is static. There a few reasons why I can come up with:
- Avoids dependency injection: If platform was an instance, everything would need to be injected with it **OR** platform could just be an instance stored inside a singleton class **OR** just be singleton itself. 
- Platform is a special class where there is always only one that should exist in the application. It could be more suited for Singleton pattern.
- Some static methods in Ably Flutter (e.g. Crypto.getDefaultParams) still need to use the method channel (state) from Platform. These methods are defined to be static in the feature spec. 

https://github.com/ably/ably-flutter/blob/78144602ab1b63568e3eaea4f79b7035e541231a/lib/src/crypto/src/crypto.dart#L18-L40

Your comment prompted me to create this PR to attempt to solve this problem by using the Singleton pattern and making the state of Platform private, however the last final issue: I can't make some fields that should be internal to the class private because they are **being used in tests.**

What do you think of https://github.com/ably/ably-flutter/pull/235 ?

Reply:
I'm resolving this conversation on @ikbalkaya's behalf, given this is being addressed (as @ben-xD states) under #235.